pkg/controller/options.go

```diff
@@ -57,6 +58,9 @@ type Options struct {
 
 	// Features that should be enabled.
 	Features *feature.Flags
+
+	// TLSConfig for controllers to support External Secret Stores
+	TLSConfig *tls.Config
```

Comment:
Do we expect most/all Crossplane controllers to support external secret stores? Is there a chance controllers will need (separate) TLS configs for other features in future?

Reply:
> Do we expect most/all Crossplane controllers to support external secret stores?
Yes, I believe so.

> Is there a chance controllers will need (separate) TLS configs for other features in future?
Yes, I would like to suggest moving this under a dedicated type for ESS.